apis/r/R/utils-readerTransformers.R

```diff
@@ -33,25 +33,33 @@ arrow_table_to_sparse <- function(tbl, repr = c("C", "T", "R"), shape = NULL, ze
   # If needed, user can then explicitly ask the shim for the underlying
   # sparseMatrix using `as.one.based()`.
 
-  soma_dim_0_one_based <- 1 + as.numeric(tbl$GetColumnByName("soma_dim_0"))
-  soma_dim_1_one_based <- 1 + as.numeric(tbl$GetColumnByName("soma_dim_1"))
-
-  soma_data <- as.numeric(tbl$GetColumnByName("soma_data"))
-
   if (is.null(shape)) {
-      shape <- c(max(soma_dim_0_one_based), max(soma_dim_1_one_based))
+    shape <- c(max(tbl$soma_dim_0)$as_vector(), max(tbl$soma_dim_1)$as_vector())
   }
 
-  if(any(shape > .Machine$integer.max)) {
-      stop("The shape of the array is larger than supported by Matrix::sparseMatrix",
-           call. = FALSE)
+  if (any(shape > .Machine$integer.max)) {
+    stop("'shape' must not exceed '.Machine$integer.max'.", call. = FALSE)
+  }
+
+  exceedsInt32Limit <- (
+    tbl$soma_dim_0 >= .Machine$integer.max | tbl$soma_dim_1 >= .Machine$integer.max
+  )
```

Comment:
One thing this PR changes is the removal of `nnz <= .Machine$integer.max`, which is important as sparse matrices cannot have more than 2^31-1 non-zero values. Can we add a check here for that (`length(tbl$soma_data) > .Machine$integer.max` I think)

Reply:
Good catch. I removed the assertion from `SOMASparseNDArray$read()` because it was blocking `SOMASparseNDArray$read()$tables()`, which is perfectly acceptable.